### DISCUSSION TITLE
PrismaClient#rawQuery not working

### DISCUSSION THREAD

**Discussion by shainegordon** [prisma/prisma#2664] [Q&A] [answered]
Currently using "@prisma/client": "2.0.0-beta.8" + Postgres 11.7

I cannot seem to get `rawQuery` to work

```
const clients = await prisma.client.findMany(); // this works fine as expected
console.log('clients', clients);
const queryResult = await prisma.queryRaw('select * from Client'); // this throws an error, see below
console.log('queryResult', queryResult);
```

Error:
```
PrismaClientKnownRequestError: Raw query failed. Code: `42P01`. Message: `db error: ERROR: relation "client" does not exist`
    at PrismaClientFetcher.request (./node_modules/@prisma/client/src/runtime/getPrismaClient.ts:762:15)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  code: 'P2010',
  meta: {
    code: '42P01',
    message: 'db error: ERROR: relation "client" does not exist'
  }
}

```

I am obviously doing something subtly wrong

**Comment by pantharshit00** [comment, marked as answer]
Hi @shainegordon 

Try doing:
```js
const queryResult = await prisma.queryRaw('select * from "public"."Client"');
```

**Reply by shainegordon** [reply]
Thank you!!, that works

**Reply by pantharshit00** [reply]
Nice, glad I can help. Remember postgres automatically lowercase your queries so you will need to quote them. Also, I added the schema `"public".` just to be explicit which is a good practice. 

So anytime you queryRaw, remember to quote your tables.

**Reply by Unsullied-Coder** [reply]
Thank you!

**Reply by aidantorrence** [reply]
why is this not in the doc 😩

**Reply by janpio** [reply]
Feel free to suggest adding this to the appropriate location in our docs (possibly raw API docs?) here: https://github.com/prisma/docs/issues/new/choose

**Reply by Pastoray** [reply]
this helped me notice the bug i had but good to know postgres does that! thank you

**Reply by IArnaut2** [reply]
Why no just make it work without `public`? I've already put that schema in the .env file:
```ENV
DATABASE_URL="postgresql://postgres:root@localhost:5432/realestateexpress?schema=public"
```

**Reply by IArnaut2** [reply]
That didn't work.
```TS
return prisma.$queryRawUnsafe(`
  SELECT *
  FROM "public"."listing"
  LEFT JOIN Apartment ON Apartment.id = Listing.apartmentId
`);
```
Error:
```
Raw query failed. Code: `42P01`. Message: `relation "public.listing" does not exist`
```

**Comment by ryands17** [comment]
Hey @shainegordon :wave: 
This would help 
```sql
const queryResult = await prisma.queryRaw('select * from "Client"');
```

**Comment by sgentile** [comment]
throws and 'unwrap' error (postgres)

**Reply by ryands17** [reply]
Are you fetching fields that aren't supported by Prisma yet? Could you share more about your schema and raw query so I can have a look.

**Comment by ludralph** [comment]
Hi @shainegordon,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!

**Comment by Serpentarius13** [comment]
I would love this to be a default behaviour (Adding "public" automatically) instead of writing it myself. I have schema=public param in connection string already

**Reply by IArnaut2** [reply]
What a shortcoming from Prisma.
